spark3/src/main/java/org/apache/iceberg/spark/DataCompactionJobCoordinator.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.FileScanTask;
+import org.apache.iceberg.HasTableOperations;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableOperations;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.util.Pair;
+
+public class DataCompactionJobCoordinator {
+
+  private static final Map<Pair<String, String>, List<FileScanTask>> TASKS = Maps.newHashMap();
+  private static final Map<Pair<String, String>, List<DataFile>> NEW_DATA_FILES = Maps.newHashMap();
+
+  private DataCompactionJobCoordinator() {
+  }
+
+  public static String stageJob(Table table, List<FileScanTask> tasks) {
+    String jobID = UUID.randomUUID().toString();
+    Pair<String, String> id = toID(table, jobID);
+    TASKS.put(id, tasks);
+    return jobID;
+  }
+
+  public static List<FileScanTask> getTasks(Table table, String jobID) {
```

Comment:
Nit: `get` in a public method name. Is there a more specific verb we can use here? Or can we omit "get" for a shorter name?